### DISCUSSION TITLE
Unable to use `snapshot.matches` because `partialStatevalue: never`.

### DISCUSSION THREAD

**Discussion by pauldesmondparker** [statelyai/xstate#5134] [Q&A] [answered]
I have not been able to produce a minimal reproducible repo, which is why I'm asking in the discussion in case someone else has seen this before and can point me in the right direction.

This react app https://stackblitz.com/edit/github-xgpyfx?file=src%2FfeedbackMachine.ts **works** and mirrors the general structure of my machine, i.e. there are parallel states at the root level.

This example is using `createActor`, `actor.start()`, `useSelector` and `snapshot.matches`.
Dependencies match on local:
```
    "@xstate/react": "5.0.0",
    "xstate": "5.19.0",
    "react": "18.3.1",
    "react-dom": "18.3.1",
    "typescript": "5.7.2",
```

In my local nextjs app when I try to use `snapshot.matches` I am getting this `partialStateValue: never` indicating that there are no states???:
```
(property) MachineSnapshotBase<TContext extends MachineContext, TEvent extends EventObject, TChildren extends Record<string, AnyActorRef | undefined>, TStateValue extends StateValue, TTag extends string, TOutput, TMeta, TStateSchema extends StateSchema = StateSchema>.matches: (partialStateValue: never) => boolean
```

There are no typescript errors on the machine definition.

Has anyone seen this or known what painful mistake I might be making?

**Comment by Andarist** [comment]
It's hard to tell if you don't share a repro case. To help out we need to see what exact code you have written

**Comment by pauldesmondparker** [comment, marked as answer]
The issue was that I was using:
```ts
// @ts-expect-error exactOptionalPropertyTypes
```
To mask this bug https://github.com/statelyai/xstate/issues/4613, which in turn masked another error which caused it not to generate the signature. I temporarily turned off `exactOptionalPropertyTypes` for the repo and found the error.
